### DISCUSSION TITLE
SimpleMeshLayer type on Pydeck

### DISCUSSION THREAD

**Discussion by OfirMazor** [visgl/deck.gl#7166] [Q&A]
Is it possible to load/display meshes files with [SimpleMeshLayer](https://deck.gl/docs/api-reference/mesh-layers/simple-mesh-layer) ?
I have tried to read an object file (.OBJ) as data for pydeck.Layer and SimpleMeshLayer as type but I get an empty Deck layer.

_*pydeck 0.7.1
*JupyterLab_

**Comment by ibgreen** [comment]
Not sure if this helps but two quick notes:

- I suspect that you'll want to provide your mesh through the `mesh` property. The `data` property accepts a "table" and will render one instance of the mesh per row. https://deck.gl/docs/api-reference/mesh-layers/simple-mesh-layer#mesh 

- The `OBJLoader` from loaders.gl would need to be bundled into pydeck to enable OBJ file parsing, sorry I am not able to check right now if it is in fact bundled. If it is, the `mesh` property should be able to just accept a URL string pointing to a mesh.

**Reply by kylebarron** [reply]
> * bundled into pydeck to enable OBJ file parsing

Not 100% sure, but I don't think it's included by default. These are the deck.gl dependencies included in default pydeck:

https://github.com/visgl/deck.gl/blob/311af2c140a0ca62f60bbe5f61a35f0b8f52af5e/modules/jupyter-widget/package.json#L31-L45
